### ISSUE TITLE
Mobile: One shared file is attached twice

### ISSUE THREAD

**Issue by wh201906** [laurent22/joplin#7311]
## Environment

Joplin version: 2.9.8
Platform: Android

## Steps to reproduce

1. Select one file in file explorer
2. Share it to Joplin
3. Check the log, the same file is attached twice
(`about to attach resource {xxx}` is printed twice, and the `Note attachments...` on desktop shows the duplicated file)

## Describe what you expected to happen

The same file should be attached only once

## Logfile

![image](https://user-images.githubusercontent.com/62299611/202473173-7d8eeced-f761-4edd-a65f-f9e2102bc41f.png)

**Comment by wh201906** [comment]
This bug is easy to reproduce and I found something which might be helpful:
1. Sharing multiple files also triggers this problem.
2. Attaching a file from the note menu doesn't trigger this problem.
3. Joplin doesn't print `Received shared data`, `about to attach resource {xxx}` when I use the `attach file` option.
4. `Received shared data` is printed only once, but `about to attach resource {xxx}`, `Got type` and `Got file` is printed twice.

**Comment by laurent22** [comment]
> 4\. `Received shared data` is printed only once, but `about to attach resource {xxx}`, `Got type` and `Got file` is printed twice.

That could mean that the Note screen is mounted, unmounted, and mounted again, because that logic for "about to attach resource" happens in componentDidMount().

**Comment by wh201906** [comment]
Is there any way to resolve this? I did find the code in `componentDidMount()`, but I don't know what to do next.
https://github.com/laurent22/joplin/blob/7d135730bf6f313aaddb488729353b93c4130b7a/packages/app-mobile/components/screens/Note.tsx#L422
https://github.com/laurent22/joplin/blob/7d135730bf6f313aaddb488729353b93c4130b7a/packages/lib/components/shared/note-screen-shared.js#L232

**Comment by laurent22** [comment]
Maybe it's possible to find the reason why it's mounting the component twice, and preventing it from doing so? Otherwise I guess we could process the sharing logic with a small delay, so that if the component is quickly mounted then unmounted nothing happens.

**Comment by wh201906** [comment]
> Maybe it's possible to find the reason why it's mounting the component twice, and preventing it from doing so?

I think this one is better, but I don't know how to debug Joplin on mobile platform. I can only modify the code to print something, and it takes a lot of time.

**Comment by wh201906** [comment]
Please keep this open.

**Comment by KshitizSareen** [comment]
Hey, Im trying to reproduce this issue, but im not able to. I attached a file from the file explorer, and one note was created with the file in it. I also saw the log and I saw about to attach resource {xxx} only once.
<img width="567" alt="Screenshot 2022-12-28 at 5 12 26 PM" src="https://user-images.githubusercontent.com/23134343/209890936-6ebf4c10-506f-48dc-b09d-166edcb7ac93.png">

**Comment by wh201906** [comment]
Thanks for your testing.
I wonder which version you are using.

**Comment by KshitizSareen** [comment]
Im using Joplin 2.10.0.
Im using an Android Emulator. Its a Pixel 6 Pro with Android 11.0.

**Comment by wh201906** [comment]
What about the version of Joplin?

**Comment by wh201906** [comment]
@KshitizSareen Sorry for getting back to you so late. I didn't noticed that you posted the version by editing the comment.

I can reproduce this problem on Joplin 2.9.8. When I try to reproduce it on 2.10.4, the share function doesn't work at all.
Not sure if it's a bug.

**Comment by laurent22** [comment]
Duplicate of #7791

**Comment by wh201906** [comment]
It was resolved in v2.10.7
